src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/MarkDuplicatesSparkIntegrationTest.java

```diff
@@ -175,6 +174,308 @@ public void testMarkDuplicatesSparkIntegrationTestLocal(
+            for ( GATKRead read : outputReads ) {
+                ++totalReads;
+
+                if ( read.isDuplicate() ) {
+                    ++duplicateReads;
+                }
+            }
+        }
+
+        Assert.assertEquals(totalReads, totalExpected - dupsExpected, "Wrong number of reads in output BAM");
+        Assert.assertEquals(duplicateReads, 0, "Wrong number of duplicate reads in output BAM");
+
+        final MetricsFile<GATKDuplicationMetrics, Comparable<?>> metricsOutput = new MetricsFile<>();
+        try {
+            metricsOutput.read(new FileReader(metricsFile));
+        } catch (final FileNotFoundException ex) {
+            System.err.println("Metrics file not found: " + ex);
+        }
+        final List<GATKDuplicationMetrics> nonEmptyMetrics = metricsOutput.getMetrics().stream().filter(
+                metric ->
+                        metric.UNPAIRED_READS_EXAMINED != 0L ||
+                                metric.READ_PAIRS_EXAMINED != 0L ||
+                                metric.UNMAPPED_READS != 0L ||
+                                metric.UNPAIRED_READ_DUPLICATES != 0L ||
+                                metric.READ_PAIR_DUPLICATES != 0L ||
+                                metric.READ_PAIR_OPTICAL_DUPLICATES != 0L ||
+                                (metric.PERCENT_DUPLICATION != null && metric.PERCENT_DUPLICATION != 0.0 && !Double.isNaN(metric.PERCENT_DUPLICATION)) ||
+                                (metric.ESTIMATED_LIBRARY_SIZE != null && metric.ESTIMATED_LIBRARY_SIZE != 0L)
+        ).collect(Collectors.toList());
+
+        // Assert that the metrics haven't changed at all
+        Assert.assertEquals(nonEmptyMetrics.size(), metricsExpected.size(),
+                "Wrong number of metrics with non-zero fields.");
+        for (int i = 0; i < nonEmptyMetrics.size(); i++ ){
+            final GATKDuplicationMetrics observedMetrics = nonEmptyMetrics.get(i);
+            List<?> expectedList = metricsExpected.get(observedMetrics.LIBRARY);
+            Assert.assertNotNull(expectedList, "Unexpected library found: " + observedMetrics.LIBRARY);
+            Assert.assertEquals(observedMetrics.UNPAIRED_READS_EXAMINED, expectedList.get(0));
+            Assert.assertEquals(observedMetrics.READ_PAIRS_EXAMINED, expectedList.get(1));
+            Assert.assertEquals(observedMetrics.UNMAPPED_READS, expectedList.get(2));
+            Assert.assertEquals(observedMetrics.UNPAIRED_READ_DUPLICATES, expectedList.get(3));
+            Assert.assertEquals(observedMetrics.READ_PAIR_DUPLICATES, expectedList.get(4));
+            Assert.assertEquals(observedMetrics.READ_PAIR_OPTICAL_DUPLICATES, expectedList.get(5));
+            Assert.assertEquals(observedMetrics.PERCENT_DUPLICATION, expectedList.get(6));
+
+            //Note: IntelliJ does not like it when a parameter for a test is null (can't print it and skips the test)
```

Comment:
I'm not sure this comment is true?  ImmutableList.of doesn't like nulls but I didn't think intellij had any trouble with it.

Reply:
Not my comment, was originally adams